packages/react-dom/src/test-utils/ReactTestUtilsAct.js

```diff
@@ -85,17 +85,19 @@ let actingUpdatesScopeDepth = 0;
 
 function act(callback: () => Thenable) {
   let previousActingUpdatesScopeDepth = actingUpdatesScopeDepth;
+  let previousActingUpdatesSigil;
   actingUpdatesScopeDepth++;
+  // we use the function flushPassiveEffects directly as the sigil,
+  // since it's unique to a renderer
   if (__DEV__) {
-    ReactShouldWarnActingUpdates.current = true;
+    previousActingUpdatesSigil = ReactActingRendererSigil.current;
+    ReactActingRendererSigil.current = flushPassiveEffects;
```

Comment:
Could you use a separate export instead? I find this unnecessarily confusing for no benefit, given that it's dev-only.